Debugger source lookups

DavidKarlas opened the pull request:
Card:
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/500134

This pull request adds support also for Portable .pdb(beside .mdb).
Adds UI familiar to VS2017 developer when source is missing instead going to Diasassembly view directly:
VSfM:
![image](https://user-images.githubusercontent.com/774791/31017213-1b77c3fc-a528-11e7-9916-4155fe8cace2.png)
VS2017:
![image](https://user-images.githubusercontent.com/774791/31016913-ef4b59f2-a526-11e7-8423-9fba1cf0b9d3.png)

It also adds options to manage "Debug Sources Files" into Solution options
![image](https://user-images.githubusercontent.com/774791/31016455-0d1dfc84-a525-11e7-9db7-7fb02dc2f599.png)

DavidKarlas commented:
@vancura Can you review 1st image?
3rd image is same component as in Preferences for something else:
![image](https://user-images.githubusercontent.com/774791/31017020-642cf1a4-a527-11e7-8b13-884d16cff9e2.png)

mhutch commented:
Looking good!

This list of search locations should probably be fullscreen. I'd also suggest adding some text like this to the first screen:

You can manage the locations used to find source files in the Solution Options.

DavidKarlas commented:
Fixed to use full options window and added text explaining lookup locations can be managed in solution options.

![image](https://user-images.githubusercontent.com/774791/31070525-fe842e6c-a760-11e7-9051-f7595cf49ec9.png)

![image](https://user-images.githubusercontent.com/774791/31070632-53796c98-a761-11e7-9d18-c5bc7a986c83.png)

vancura commented:
Please simplify, nearly all of this text is useless:

![screen shot 2017-10-02 at 17 33 40](https://user-images.githubusercontent.com/4982/31085587-e349519a-a797-11e7-89fe-2ab4a27103d0.png)

Let me know when you fix this so I can re-review. Thanks!

DavidKarlas commented:
@vancura 
![image](https://user-images.githubusercontent.com/774791/31118694-9981d964-a82e-11e7-8090-81450f52d44d.png)

vancura commented:
Perfect! Can you please show me how it will look when there are multiple buttons side by side?

DavidKarlas commented:
Currently there are no other actions, not sure in which case there are in VS, hence I made dummy buttons
![image](https://user-images.githubusercontent.com/774791/31118883-54b12852-a82f-11e7-83c9-5cfa8270609d.png)

vancura commented:
Amazing! Thanks!

Therzok commented on main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/NoSourceView.cs:
```diff
@@ -55,21 +55,25 @@ Widget CreateContent (bool disassemblyNotSupported)
-				box.PackStart (browseAndFindLabel);
-				box.PackStart (new Label (GettextCatalog.GetString ("You can manage the locations used to find source files in the Solution Options.")));
+				var actionsBox = new HBox ();
+				var buttonBrowseAndFind = new Button (GettextCatalog.GetString ($"Browse and find {fileName}"));
```
This won't get picked up by the regex parser.

Therzok commented on main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/NoSourceView.cs:
```diff
@@ -80,7 +84,7 @@ Widget CreateContent (bool disassemblyNotSupported)
 			if (!disassemblyNotSupported) {
 				var labelDisassembly = new Label ();
-				labelDisassembly.Markup = GettextCatalog.GetString ("You can {0} in the Disassembly window.", $"<a href=\"clicked\">{GettextCatalog.GetString ("view disassembly")}</a>");
+				labelDisassembly.Markup = GettextCatalog.GetString ("View disassembly in the {0}", $"<a href=\"clicked\">{GettextCatalog.GetString ("Disassembly Tab")}</a>");
```
Not sure if the 2nd string will be picked up by the regex parser. Better split this into a non-interpolated string, same with manageLookupsLabel above.

decriptor commented:
`Browse and find` feels redundant to me. I'm not sure if just `Find File.cs` or `Locate File.cs` is better?
What about instead of `Manage the locations...`, `Manage source file locations in Solution Options` (not sure if that changes the meaning)?

Instead of `View disassembly in the Disassembly Tab` maybe just `View assembly` with that text being the link. Would it be inferred that it'll be disassembled?

DavidKarlas commented:
I prefer to stay closer to VS2017 text...

slluis commented:
@DavidKarlas please fix conflicts
